### DISCUSSION TITLE
could I install extension from visx through theia-ide docker?

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-theia/theia#13814] [Q&A] [answered]
hi,
    I want to deploy my theia-ide based application to docker, I download eclipse-theia/theia-blueprint/theia-ide:1.50.100
![image](https://github.com/eclipse-theia/theia/assets/4537809/26752696-a2de-4d2a-a5c5-d6776f33889f)
and docker runs it.
   My problem is that if I want to install a VISX extension from my local machine, how could I do it, the browser can not open my local machine driver
![image](https://github.com/eclipse-theia/theia/assets/4537809/38fa865a-e6a4-4c73-a436-8d79e3f67285)
![image](https://github.com/eclipse-theia/theia/assets/4537809/28291c49-0585-40f3-8d35-dd5543d34769)

**Comment by msujew** [comment, marked as answer]
Hey @wss29,

you can drag & drop the `.vsix` the file from your file system to the explorer of the app. Alternatively use the `Upload Files...` command. You can then select the uploaded vsix file to install your extension.

**Comment by wss29** [comment]
Hi @msujew, 
Thank you for your feedback. The drag & drop feature can also upload folders, which is great! However, the upload files command seems to only upload individual files. It would be helpful to have a command that can upload entire folders in the future.
